sup3r/pipeline/forward_pass.py

```diff
@@ -31,6 +32,413 @@
 logger = logging.getLogger(__name__)
 
 
+class ForwardPassSlicer:
```

Comment:
Nice i like this delegation of slicing responsibility to a single class and the encapsulation with properties is great!

Reply:
Thanks!